src/snowflake/connector/ocsp_snowflake.py

```diff
@@ -1100,6 +1116,20 @@ def validate_by_direct_connection(self, issuer, subject, telemetry_data, hostnam
                         # TODO - Remove this potentially broken OCSP Response / SSD
                         raise op_er
                 else:
+                    if op_er.errno == ER_INVALID_OCSP_RESPONSE_CODE:
+                        telemetry_data.set_event_sub_type('OCSPResponseLoadFailure')
+                    elif op_er.errno == ER_OCSP_RESPONSE_STATUS_UNSUCCESSFUL:
+                        telemetry_data.set_event_sub_type("OCSPResponseStatusUnsuccessful")
+                    elif op_er.errno == ER_OCSP_RESPONSE_ATTACHED_CERT_INVALID:
+                        telemetry_data.set_event_sub_type("OCSPResponseAttachedCertInvalid")
+                    elif op_er.errno == ER_OCSP_RESPONSE_ATTACHED_CERT_EXPIRED:
+                        telemetry_data.set_event_sub_type("OCSPResponseAttachedCertExpired")
+                    elif op_er.errno == ER_OCSP_RESPONSE_INVALID_SIGNATURE:
+                        telemetry_data.set_event_sub_type("OCSPResponseSignatureInvalid")
+                    elif op_er.errno == ER_OCSP_RESPONSE_INVALID_EXPIRY_INFO_MISSING:
+                        telemetry_data.set_event_sub_type("OCSPResponseExpiryInfoMissing")
+                    elif op_er.errno == ER_OCSP_RESPONSE_EXPIRED:
+                        telemetry_data.set_event_sub_type("OCSPResponseExpired")
                     raise op_er
```

Comment:
what if no error is matched? no sub type?

Reply:
The goal was to populate telemetry data with the right event_sub_type. Part of the OCSP revocation check failed exceptions are raised in ocsp_asn1crypto.py. I had two choices, either expose telemetry data to ocsp_asn1crypto or use the err no of the revocation check exception raised in it to fill the telemetry data in the calling function itself. The point being I have added new error codes each of which corresponds to an event_sub_type in ocsp_snowflake.py. Removing the case of no error number is matched. In the future if we introduce new error numbers we will create a corresponding event_sub_type to maintain the granularity. Worse case if someone forgets to add a corresponding sub type, then it would show up as none which is the default.

Reply:
ok, sound like the sub type is a string representation of errno for OOB. If you are concern about missing corresponding sub type for the errno, you could do this:
- Create a map/dict data structure from errno to sub type
- Create a test case to attempt to map all of the existing OCSP errno and ensure the sub type is returned. If not returned, it means no sub type is defined, so the developer will notice he/she has to add a subtype whenever a new OCSP errno is introduced.

Reply:
That's a good idea.  A few things: 
1. At first I did not think that all error numbers need to be mapped to event sub types since this mapping was just introduced to not have to send telemetry_data object to oscp_asn1crypto. So only the error numbers being used in ocsp_asn1crypto really needed to be mapped. That said I don't see any harm in mapping all ocsp related error number to an event sub type. 
2. About your idea to avoid a potential lapse by a developer. The developer will still need to update the map with the new error number after introducing it in the errorcode.py so that the test knows a new error number has been introduced.